### PR TITLE
[codex] connect DAG and Trinity dispatcher diagnostics

### PR DESCRIPTION
**Pull request by pbjustin**
## Summary

- Adds deterministic dispatcher diagnostics metadata for DAG, Trinity, workers, control-plane, and MCP bindings.
- Classifies natural-language DAG execution/diagnostic prompts as DAG control-plane traffic with DAG-specific endpoint guidance.
- Updates the GPT route OpenAPI contract and focused tests for dispatcher diagnostics and Trinity control-plane leak protection.

## Root Cause

The core `/gpt/{gptId}` dispatcher was healthy, but DAG/Trinity diagnostic prompts were routed through generic runtime-inspection classification. That produced `CONTROL_PLANE_REQUIRES_DIRECT_ENDPOINT` and left the dispatcher diagnostics response without enough subsystem binding metadata to verify the DAG/Trinity connection.

## Validation

- `node -e "JSON.parse(require('fs').readFileSync('contracts/custom_gpt_route.openapi.v1.json','utf8')); console.log('openapi json ok')"`
- `node scripts/run-jest.mjs --runTestsByPath tests/runtime-diagnostics.route.test.ts tests/gpt-router-auth-logging.test.ts tests/trinity-writing-pipeline.test.ts tests/gpt-router-universal-dispatch.test.ts tests/dispatcher-priority.route.test.ts --coverage=false`
- `npm run type-check`

Note: `npm test -- --runTestsByPath ...` also had all selected tests pass, but exited non-zero because the repo enforces full coverage thresholds even on focused path runs.

### CODE REVIEW

File: contracts/custom_gpt_route.openapi.v1.json [hunk 1]
```diff
@@ -231,6 +237,23 @@
...
+                      "error": {
+                        "code": "DAG_CONTROL_REQUIRES_DIRECT_ENDPOINT",
+                        "message": "DAG execution must use /api/arcanos/dag/*, POST /mcp, or POST /dispatch with target='dag'."
+                      },
+                      "canonical": {
+                        "mcp": "/mcp",
+                        "dispatch": "/dispatch",
+                        "dagRuns": "/api/arcanos/dag/runs/{runId}",
+                        "dagTrace": "/api/arcanos/dag/runs/{runId}/trace"
+                      }
```
[the remark this answers is not in the document]
Comment (pbjustin) [reply contracts/custom_gpt_route.openapi.v1.json:256]: Addressed in eaa12ec5. The OpenAPI DAG control rejection example now includes dagCapabilities, uses dagRuns for the collection endpoint, adds dagRunStatus for the item template, and keeps dagTrace aligned with GptDispatcherDiagnosticsResponse.

File: src/platform/runtime/writingPlaneContract.ts [hunk 2]
```diff
@@ -169,6 +175,31 @@ function buildUnsupportedControlActionCanonical() {
...
 }
 
+function buildDagControlCanonical() {
+  return {
+    mcp: '/mcp',
+    dispatch: '/dispatch',
+    dagRuns: '/api/arcanos/dag/runs/{runId}',
+    dagTrace: '/api/arcanos/dag/runs/{runId}/trace',
+  };
+}
```
[the remark this answers is not in the document]
Comment (pbjustin) [reply src/platform/runtime/writingPlaneContract.ts:187]: Addressed in eaa12ec5. buildDagControlCanonical now matches the dispatcher diagnostics naming: dagCapabilities, dagRuns as /api/arcanos/dag/runs, dagRunStatus as /api/arcanos/dag/runs/{runId}, and dagTrace.

File: tests/gpt-router-auth-logging.test.ts [hunk 3]
```diff
@@ -727,6 +727,45 @@ describe('gpt router auth logging', () => {
...
+      error: {
+        code: 'DAG_CONTROL_REQUIRES_DIRECT_ENDPOINT',
+        message: "DAG execution must use /api/arcanos/dag/*, POST /mcp, or POST /dispatch with target='dag'.",
+      },
+      canonical: {
+        mcp: '/mcp',
+        dispatch: '/dispatch',
+        dagRuns: '/api/arcanos/dag/runs/{runId}',
+        dagTrace: '/api/arcanos/dag/runs/{runId}/trace',
+      },
```
[the remark this answers is not in the document]
Comment (pbjustin) [reply tests/gpt-router-auth-logging.test.ts:763]: Addressed in eaa12ec5. The route expectations now assert the expanded DAG canonical endpoint structure for both explicit embedded DAG actions and natural-language DAG diagnostic prompts.
